src/ArrowFunctions.cpp

```diff
@@ -333,29 +333,35 @@ int cpp_readColumnByName(const char* filename, void* chpl_arr, const char* colna
         parquet::FloatReader* reader =
           static_cast<parquet::FloatReader*>(column_reader.get());
         startIdx -= reader->Skip(startIdx);
-
-        float* tmpArr = (float*)malloc(batchSize * sizeof(float));
+        
         while (reader->HasNext() && i < numElems) {
-          if((numElems - i) < batchSize)
-            batchSize = numElems - i;
+          float value;
           // Can't read directly into chpl_ptr because it is a double
-          (void)reader->ReadBatch(batchSize, nullptr, nullptr, tmpArr, &values_read);
-          for (int64_t j = 0; j < values_read; j++)
-            chpl_ptr[i+j] = (double)tmpArr[j];
-          i+=values_read;
+          (void)reader->ReadBatch(1, &definition_level, nullptr, &value, &values_read);
```

Comment:
Haven't had a chance to do a full review, but just noting at a quick glance that this is going to result in a significant performance hit to only read one value at a time as opposed to `batchSize` at a time.

Reply:
Discussed this with @bmcdonald3. After reviewing our current options, there is not an obvious alternative for this to work properly. We are going to continue with this resolution and add an additional issue to pursue this in a more efficient manor.